### PR TITLE
Introduce PurgeKernels class (bsc#1155198)

### PR DESCRIPTION
**Pull request by bzeller**
Adds libzypp API to mark all obsolete kernels according to the existing
purge-kernel script rules

### CODE REVIEW

File: tests/repo/RepoLicense_test.cc [hunk 1]
```diff
@@ -11,15 +11,24 @@ using namespace zypp;
 using std::cout;
 using std::endl;
 
-TestSetup test( Arch_x86_64 );
+
+static std::unique_ptr<TestSetup> testSetup;
```
Comment (mlandres) [inline tests/repo/RepoLicense_test.cc]: Basically right, but I'd prefer a solution that requires less syntactical changes in the existing code (no var rename, no var to ptr).
The `std::unique_ptr` can be used as [`_pimpl` for TestSetup](https://github.com/openSUSE/libzypp/blob/ma-draft/tests/lib/TestSetup.h). A testcase intro could then look like [this](https://github.com/openSUSE/libzypp/blob/ma-draft/tests/repo/RepoLicense_test.cc#L14). Usage of `test` in existing testcase would not need to be changed.

Comment (bzeller) [reply tests/repo/RepoLicense_test.cc]: Done

File: zypp/PurgeKernels.cc [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+    const str::regex specRegex( "^(latest|oldest)([+-][0-9]+)?$", str::regex::match_extended );
+
+    const unsigned tokenGrp = 1; //index of the group matching the token
+    const unsigned modifierGrp = 2; //index of the group matching the offset modifier
+
+
+    MIL << "Parsing keep spec: " << _keepSpec << std::endl;
+
+    std::vector<std::string> words;
+    str::split( _keepSpec, std::back_inserter(words), "," );
```
Comment (mlandres) [inline zypp/PurgeKernels.cc]: I'd relax the parser to trim the words. Actually I'd enhance str::split to accept an optional `Trim` argument:
```
    template<class TOutputIterator>
      unsigned split( const C_Str & line_r, TOutputIterator result_r, const C_Str & sepchars_r = " \t", const Trim trim_r = NO_TRIM )
    template<class TOutputIterator>
      unsigned split( const C_Str & line_r, TOutputIterator result_r, const Trim trim_r )
      { return split( line_r, result_r, " \t", trim_r ); }
```
May cherry-pick b4343234febc303c6f655e6b2905c4346cb49b11

Comment (bzeller) [reply zypp/PurgeKernels.cc]: Done

**Pull request merged**